Makefile

```diff
@@ -23,18 +23,14 @@ SHELL:=/usr/bin/env bash
 #
 # Go.
 #
-GO_VERSION ?= 1.24.11
-GO_DIRECTIVE_VERSION ?= 1.24.0
+GO_VERSION ?= 1.25.5
+GO_DIRECTIVE_VERSION ?= 1.25.0
 GO_CONTAINER_IMAGE ?= docker.io/library/golang:$(GO_VERSION)
 
 # Ensure correct toolchain is used
 GOTOOLCHAIN = go$(GO_VERSION)
 export GOTOOLCHAIN
 
-# Required with Go 1.24 to enable usage of synctest in unit tests
```

Comment:
We also have to drop this in .github/workflows/pr-golangci-lint.yaml

We also have to adjust some code, please pick up the changes from https://github.com/kubernetes-sigs/cluster-api/pull/13159/files#diff-9dd6baa9eadbe9f208056db3d12c403cac506f53841e987cf67fd9b9391429f0